src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -30,18 +32,16 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```

Comment:
```suggestion
      throw new IllegalStateException("numberOfTorpedos");
```
Mivel itt valami állapotról van szó, lehetne IllegalStateException-t is dobni...